code/game/gamemodes/changeling/powers/mutations.dm

```diff
@@ -174,6 +177,180 @@
+	weapon_type = /obj/item/weapon/gun/magic/tentacle
+	weapon_name_simple = "tentacle"
+	silent = TRUE
+
+/obj/item/weapon/gun/magic/tentacle
+	name = "tentacle"
+	desc = "A fleshy tentacle that can stretch out and grab things or people."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "tentacle"
+	item_state = "tentacle"
+	flags = ABSTRACT | NODROP | NOBLUDGEON
+	w_class = WEIGHT_CLASS_HUGE
+	ammo_type = /obj/item/ammo_casing/magic/tentacle
+	fire_sound = 'sound/effects/splat.ogg'
+	force = 0
+	max_charges = 1
+	throwforce = 0 //Just to be on the safe side
+	throw_range = 0
+	throw_speed = 0
+
+/obj/item/weapon/gun/magic/tentacle/New(location,silent)
+	..()
+	if(ismob(loc))
+		if(!silent)
+			loc.visible_message("<span class='warning'>[loc.name]\'s arm starts stretching inhumanly!</span>", "<span class='warning'>Our arm twists and mutates, transforming it into a tentacle.</span>", "<span class='italics'>You hear organic matter ripping and tearing!</span>")
+		else
+			to_chat(loc, "<span class='notice'>You prepare to extend a tentacle.</span>")
+
+/obj/item/weapon/gun/magic/tentacle/dropped()
+	. = ..()
+	qdel(src)
+
+/obj/item/weapon/gun/magic/tentacle/shoot_with_empty_chamber(mob/living/user as mob|obj)
+	to_chat(user, "<span class='warning'>The [name] is not ready yet.<span>")
+
+/obj/item/weapon/gun/magic/tentacle/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] coils [src] tightly around \his neck! It looks like \he's trying to commit suicide.</span>")
+	return (OXYLOSS)
+
+/obj/item/ammo_casing/magic/tentacle
+	name = "tentacle"
+	desc = "a tentacle."
+	projectile_type = /obj/item/projectile/tentacle
+	caliber = "tentacle"
+	icon_state = "tentacle_end"
+	var/obj/item/weapon/gun/magic/tentacle/gun //the item that shot it
+
+/obj/item/ammo_casing/magic/tentacle/New(obj/item/weapon/gun/magic/tentacle/tentacle_gun)
+	gun = tentacle_gun
+	..()
+
+/obj/item/ammo_casing/magic/tentacle/Destroy()
+	gun = null
+	return ..()
+
+/obj/item/projectile/tentacle
+	name = "tentacle"
+	icon_state = "tentacle_end"
+	pass_flags = PASSTABLE
+	damage = 0
+	damage_type = BRUTE
+	range = 8
+	hitsound = 'sound/weapons/thudswoosh.ogg'
+	var/chain
+	var/obj/item/ammo_casing/magic/tentacle/source //the item that shot it
+
+/obj/item/projectile/tentacle/New(obj/item/ammo_casing/magic/tentacle/tentacle_casing)
+	source = tentacle_casing
+	..()
+
+/obj/item/projectile/tentacle/fire(setAngle)
+	if(firer)
+		chain = firer.Beam(src, icon_state = "tentacle", time = INFINITY, maxdistance = INFINITY, beam_sleep_time = 1)
+	..()
+
+/obj/item/projectile/tentacle/proc/reset_throw(mob/living/carbon/human/H)
+	if(H.in_throw_mode)
+		H.throw_mode_off() //Don't annoy the changeling if he doesn't catch the item
+
+/mob/proc/tentacle_grab(mob/living/carbon/C)
+	if(Adjacent(C))
+		var/obj/item/weapon/grab/G = C.grabbedby(src,1)
+		if(istype(G))
+			G.state = GRAB_AGGRESSIVE //Instant aggressive grab
+
+/mob/proc/tentacle_stab(mob/living/carbon/C)
```

Comment:
I tried everything to make the projectile procs work, but they literally never fired regardless of how i setup the callback, addtimer, spawned call, the projectile always managed to get deleted beforehand. This was the easiest solution.